plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java

```diff
@@ -0,0 +1,420 @@
+        mapCapabilities.put(DataStoreCapabilities.CAN_REVERT_VOLUME_TO_SNAPSHOT.toString(), Boolean.FALSE.toString());
+
+        mapCapabilities.put(DataStoreCapabilities.VOLUME_SNAPSHOT_QUIESCEVM.toString(), Boolean.FALSE.toString());
+
+        return mapCapabilities;
+    }
+
+    @Override
+    public DataTO getTO(DataObject data)
+    {
+        return null;
+    }
+
+    @Override
+    public DataStoreTO getStoreTO(DataStore store)
+    {
+        return null;
+    }
+
+    @Override
+    public ChapInfo getChapInfo(DataObject dataObject)
+    {
+        return null;
+    }
+
+    @Override
+    public boolean grantAccess(DataObject dataObject, Host host, DataStore dataStore)
+    {
+        return false;
+    }
+
+    @Override
+    public void revokeAccess(DataObject dataObject, Host host, DataStore dataStore)
+    {
+    }
+
+    @Override
+    public long getUsedBytes(StoragePool storagePool)
+    {
+        return 0;
+    }
+
+    @Override
+    public long getUsedIops(StoragePool storagePool)
+    {
+        return 0;
+    }
+
+    @Override
+    public long getDataObjectSizeIncludingHypervisorSnapshotReserve(DataObject dataObject, StoragePool pool)
+    {
+        return dataObject.getSize();
+    }
+
+    @Override
+    public long getBytesRequiredForTemplate(TemplateInfo templateInfo, StoragePool storagePool)
+    {
+        return 0;
+    }
+
+    private void deleteResourceDefinition(DataStore dataStore, String rscDefName)
+    {
+        StoragePoolVO storagePool = _storagePoolDao.findById(dataStore.getId());
+        DevelopersApi linstorApi = LinstorUtil.getLinstorAPI(storagePool.getHostAddress());
+
+        try
+        {
+            ApiCallRcList answers = linstorApi.resourceDefinitionDelete(rscDefName);
+            if (ApiCallChecker.hasError(answers))
+            {
+                for (ApiCallRc answer : answers)
+                {
+                    s_logger.error(answer.getMessage());
+                }
+                throw new CloudRuntimeException("Linstor: Unable to delete resource definition: " + rscDefName);
+            }
+        } catch (ApiException apiEx)
+        {
+            s_logger.error("Linstor: ApiEx - " + apiEx.getMessage());
+            throw new CloudRuntimeException(apiEx);
+        }
+    }
+
+    @Override
+    public void deleteAsync(
+        DataStore dataStore,
+        DataObject dataObject,
+        AsyncCompletionCallback<CommandResult> callback)
+    {
+        s_logger.debug("deleteAsync: " + dataObject.getType() + ";" + dataObject.getId());
+        String errMsg = null;
+
+        StoragePoolVO storagePool = _storagePoolDao.findById(dataStore.getId());
+
+        if (dataObject.getType() == DataObjectType.VOLUME)
+        {
+            VolumeInfo volumeInfo = (VolumeInfo) dataObject;
+
+            long storagePoolId = dataStore.getId();
+            final String rscName = "cs-" + volumeInfo.getPath();
```

Comment:
Better to define the prefix "_cs-_" as string constant in some common class, and use it across.